msg/templates/urtps/Subscriber.cpp.em

```diff
@@ -132,17 +135,27 @@ void @(topic)_Subscriber::SubListener::onSubscriptionMatched(Subscriber* sub, Ma
 
 void @(topic)_Subscriber::SubListener::onNewDataMessage(Subscriber* sub)
 {
+        std::unique_lock<std::mutex> has_msg_lock(has_msg_mutex);
+        if(has_msg.load() == true) // Check if msg has been fetched
+        {
+            has_msg_cv.wait(has_msg_lock); // Wait till msg has been fetched
+        }
+        has_msg_lock.unlock();
+        
+
         // Take data
         if(sub->takeNextData(&msg, &m_info))
         {
             if(m_info.sampleKind == ALIVE)
             {
-                // Print your structure data here.
+                std::unique_lock<std::mutex> lk(*t_send_queue_mutex);
```

Comment:
I do like using a scoped_lock as well, but I think that's only available in C++17 and PX4 code base is C++11.

Reply:
What I was thinking was something like:
```C++
{
    lock_guard lck(mtx);
    // do stuff
    // do more stuff
}
cv.notify();
```
I like this pattern because it is very clear what is with the mutex, and what is without it.

But the `unique_lock` also works fine.